### PR TITLE
feat(dato-panel): update dato panel close observables

### PR DESCRIPTION
**Pull request by shaharkazaz**
Changed close$ to afterClose and added beforeClose

BREAKING CHANGE: open function now returns an object with both afterClose and beforeClose instead of the close$ observable

### CODE REVIEW

File: playground/src/app/components-gallery/previews/panel-preview/panel-preview.component.ts [hunk 1]
```diff
@@ -18,7 +18,10 @@ export class PanelPreviewComponent implements OnInit {
   }
 
   open() {
-    this.panelService.open(PanelDemoComponent).subscribe(() => console.log('closed'));
```
Comment (bh86) [inline playground/src/app/components-gallery/previews/panel-preview/panel-preview.component.ts]: Please add an example of passing the result object back to the opener.

File: lib/src/panel/panel.service.ts [hunk 2]
```diff
@@ -17,56 +17,72 @@ import { DatoPanelComponent } from './panel.component';
...
+      });
       this.destroy$.pipe(take(1)).subscribe(() => {
         this.createPanel(content, _relativeTo, options);
       });
     } else {
       this.createPanel(content, _relativeTo, options);
     }
 
+    return {
+      beforeClose: <Result>(): Observable<DatoPanelResult<Result>> => {
```
Comment (NetanelBasal) [inline lib/src/panel/panel.service.ts]: for what we need before close?

**Comment by shaharkazaz** [comment]
The panel was refactored by @bh86 so this is no longer relevant.

**Comment by NetanelBasal** [comment]
What do u mean?

**Comment by shaharkazaz** [comment]
Amit is now opening a PR about it.

**Pull request closed without merging**